builtin/logical/pki/cert_util.go

```diff
@@ -162,6 +241,291 @@ func validateCommonNames(req *logical.Request, commonNames []string, role *roleE
 	return "", nil
 }
 
+func generateCert(b *backend,
```

Comment:
I feel like we are over threading the `FieldData` through these helpers. Many should probably take a much simpler intermediary configuration that is cleaner, has stronger typing, makes it easier to test, etc